packages/admin/src/FormSaveSplitButton.tsx

```diff
@@ -0,0 +1,52 @@
+import { FORM_ERROR } from "final-form";
+import * as React from "react";
+import { PropsWithChildren } from "react";
+import { useForm, useFormState } from "react-final-form";
+import { FormattedMessage } from "react-intl";
+
+import { SaveButton } from "./common/buttons/save/SaveButton";
+import { SplitButton } from "./common/buttons/split/SplitButton";
+import { useStackApi } from "./stack/Api";
+
+export interface FormSaveButtonProps {
+    localStorageKey?: string;
+    saving: boolean;
+    hasErrors: boolean;
+}
+
+export const FormSaveSplitButton = ({ saving, hasErrors, localStorageKey }: PropsWithChildren<FormSaveButtonProps>) => {
+    const stackApi = useStackApi();
+    const form = useForm();
+    const { pristine, hasValidationErrors, submitting } = useFormState();
+
+    return (
+        <SplitButton disabled={pristine || hasValidationErrors || submitting} localStorageKey={localStorageKey}>
+            <SaveButton
+                color={"primary"}
+                variant={"contained"}
+                saving={saving}
+                hasErrors={hasErrors}
+                onClick={() => {
+                    form.submit();
+                }}
+            >
+                <FormattedMessage id="comet.generic.save" defaultMessage="Save" />
+            </SaveButton>
+            <SaveButton
+                color={"primary"}
+                variant={"contained"}
+                saving={saving}
+                hasErrors={hasErrors != null}
```

Comment:
Form state does contain submitting and hasSubmitErrors. We could use that and avoid having to pass those as props. 

As this component can't be used without FinalForm you could even completely rely on those and remove the saving/hasErrors props.